lib/freno/client.rb

```diff
@@ -178,5 +181,20 @@ def decorated(request)
         outermost
       end
     end
+
+    def validate_and_register_decorators(decorators)
+      decorators.each do |decorator|
+        raise DecorationError if already_registered?(decorator)
```

Comment:
Giving this will be noticeable by the use if they are decorating the client wrong, I believe we should provide a descriptive error message indicating why this is failing.  What do you think?